### ISSUE TITLE
latest_testflight_build_number returns always 1

### ISSUE THREAD

**Issue by mkalmes** [fastlane/fastlane#11147]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Using `fastlane` 2.64.0 and `bundle exec fastlane run latest_testflight_build_number` returns always 1 instead of the actual value in iTunesConnect.

Updating fastlane to the latest version (2.68.2) solves this issue, but crashes in `gym` (see #10500 and CocoaPods/Xcodeproj#515).

### Captured Output

Command Used: `run latest_testflight_build_number`
<details><summary>Output/Log</summary>

```

DEBUG [2017-12-07 16:22:09.49]: Checking if there are any plugins that should be loaded...
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
INFO [2017-12-07 16:22:09.68]: --------------------------------------------
INFO [2017-12-07 16:22:09.68]: --- Step: latest_testflight_build_number ---
INFO [2017-12-07 16:22:09.68]: --------------------------------------------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------
Successfully loaded Appfile at path '/Users/marc.kalmes/Projects/ios/fastlane/Appfile'
- app_identifier: 'de.pixelhouse.chefkoch-iphone'
-------


```

</details>

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                          |
| --------------------------- | -------------------------------------------------------------- |
| OS                          | 10.13.2                                                        |
| Ruby                        | 2.3.3                                                          |
| Bundler?                    | true                                                           |
| Git                         | git version 2.14.1                                             |
| Installation Source         | ~/gemrepo/bin/fastlane                                         |
| Host                        | Mac OS X 10.13.2 (17C88)                                       |
| Ruby Lib Dir                | /System/Library/Frameworks/Ruby.framework/Versions/2.3/usr/lib |
| OpenSSL Version             | LibreSSL 2.2.7                                                 |
| Is contained                | false                                                          |
| Is homebrew                 | false                                                          |
| Is installed via Fabric.app | false                                                          |
| Xcode Path                  | ~/Applications/Xcode.app/Contents/Developer/                   |
| Xcode Version               | 9.2                                                            |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# More documentation about how to customize your build
# can be found here:
# https://docs.fastlane.tools
fastlane_version "1.109.0"

default_platform :ios

lane :test do
  scan(scheme: "Chefkoch")
end

# Fastfile actions accept additional configuration, but
# don't worry, fastlane will prompt you for required
# info which you can add here later
lane :beta do
  cocoapods

  increment_build_number(
      build_number: latest_testflight_build_number(version: get_version_number) + 1
  )

  # build your iOS app
  gym(
    scheme: "Chefkoch",
    export_method: "app-store",
    export_options: {
      provisioningProfiles: {
        "de.pixelhouse.chefkoch-iphone" => "Chefkoch iOS Distribution",
        "de.pixelhouse.chefkoch-iphone.widget" => "Chefkoch Widget Distribution"
      }
    }
  )

  # upload to Testflight
  pilot(skip_waiting_for_build_processing: true)

  # upload dSYM to Hockey
  hockey(
    api_token: <token>,
    public_identifier: <identifier>,
    upload_dsym_only: true
  )

  slack(
    slack_url: <url>
  )
end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
# The Appfile can be used to specify information that's used across all fastlane
# tools, like your username or the app's bundle identifier.
#
# For more details, check out the documentation at:
# https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Appfile.md

app_identifier "de.pixelhouse.chefkoch-iphone" # the bundle identifier of your app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.64.0  | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.0.0        |
| bundler                   | 1.16.0       |
| io-console                | 0.4.5        |
| CFPropertyList            | 2.3.5        |
| i18n                      | 0.8.4        |
| minitest                  | 5.10.2       |
| thread_safe               | 0.3.6        |
| tzinfo                    | 1.2.3        |
| activesupport             | 4.2.8        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.2        |
| babosa                    | 1.0.2        |
| claide                    | 1.0.2        |
| fuzzy_match               | 2.0.4        |
| nap                       | 1.1.0        |
| cocoapods-core            | 1.1.1        |
| cocoapods-deintegrate     | 1.0.1        |
| cocoapods-downloader      | 1.1.3        |
| cocoapods-plugins         | 1.0.0        |
| cocoapods-search          | 1.0.0        |
| cocoapods-stats           | 1.0.0        |
| netrc                     | 0.7.8        |
| cocoapods-trunk           | 1.1.2        |
| cocoapods-try             | 1.1.0        |
| colored                   | 1.2          |
| escape                    | 0.0.4        |
| fourflusher               | 2.0.1        |
| gh_inspector              | 1.0.3        |
| molinillo                 | 0.5.4        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.5.1        |
| cocoapods                 | 1.1.1        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.5        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| dotenv                    | 2.2.1        |
| excon                     | 0.59.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.13.1       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.12.2       |
| fastimage                 | 2.1.0        |
| jwt                       | 2.1.0        |
| little-plugger            | 1.1.4        |
| multi_json                | 1.12.2       |
| logging                   | 2.2.2        |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| signet                    | 0.8.1        |
| googleauth                | 0.6.2        |
| httpclient                | 2.8.3        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.1        |
| google-api-client         | 0.13.6       |
| json                      | 2.1.0        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.6.0        |
| plist                     | 3.3.0        |
| rubyzip                   | 1.2.1        |
| security                  | 0.1.3        |
| slack-notifier            | 1.5.1        |
| terminal-notifier         | 1.8.0        |
| unicode-display_width     | 1.3.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.5.1        |
| word_wrap                 | 1.0.0        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.2.8        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2017-12-07**
</details>

**Comment by andresinaka** [comment]
Same happening to me. It was working everything fine and then I started to get this message: 
`You have to specify a new version number, as there are multiple to choose from`

I specified the version number and then it's returning "1" all the time. 

Changes from apple?

**Comment by schrismartin** [comment]
I'm also running into this issue.

**Comment by lgauthier** [comment]
I'm seeing this issue as well, exactly as @andresinaka describes.

**Comment by ohayon** [comment]
I am also seeing some weird behavior here... taking a look into this now to see if there is a quick fix! 🚀

**Comment by ohayon** [comment]
So it appears that TestFlight is having some issues. I would love to say that this is happening because of their endpoints doing some weird stuff. Lets track this and see what happens when they are back up and running

**Comment by andresinaka** [comment]
thanks for the update @ohayon!

**Comment by andresinaka** [comment]
@ohayon it's back! working fine for me!

**Comment by mkalmes** [comment]
I'm still hitting the same error as described before.

Running `bundle exec fastlane run latest_testflight_build_number` results in `You have to specify a new version number, as there are multiple to choose from`.

Specifying a version number (eg. `bundle exec fastlane run latest_testflight_build_number version:2.11.1`) returns:
```
[09:19:51]: Fetching the latest build number for version 2.11.1
[09:19:51]: Latest upload is build number: 1
[09:19:51]: Result: 1
```

Using `fastlane` 2.68.2 and running `bundle exec fastlane run latest_testflight_build_number` returns the expected result:
```
[09:20:26]: Fetching the latest build number for version 2.11.1
[09:20:27]: Latest upload for version 2.11.1 is build: 2
[09:20:27]: Result: 2
```
 
But using any version newer than 2.64.0 depends on `xcodeproj` 1.5.2 or above, which will crash `gym` (see stack trace below).


<details><summary>Stack Trace</summary>

```
[09:14:39]: fastlane finished with errors
bundler: failed to load command: fastlane (/Users/marc.kalmes/gemrepo/bin/fastlane)
SystemStackError: stack level too deep
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:122:in `resolve_variable_substitution'
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:87:in `resolve_build_setting'
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:128:in `resolve_variable_substitution'
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:131:in `resolve_variable_substitution'

<truncated for better readability>

  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:128:in `resolve_variable_substitution'
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:131:in `resolve_variable_substitution'
  /Users/marc.kalmes/gemrepo/gems/xcodeproj-1.5.3/lib/xcodeproj/project/object/build_configuration.rb:87:in `resolve_build_setting'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:114:in `block (3 levels) in detect_project_profile_mapping'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:107:in `each'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:107:in `block (2 levels) in detect_project_profile_mapping'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:106:in `each'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:106:in `block in detect_project_profile_mapping'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:96:in `each'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:96:in `detect_project_profile_mapping'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/code_signing_mapping.rb:16:in `merge_profile_mapping'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/detect_values.rb:70:in `detect_selected_provisioning_profiles'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/detect_values.rb:24:in `set_additional_default_values'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym.rb:26:in `config='
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/gym/lib/gym/manager.rb:4:in `work'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/actions/build_ios_app.rb:65:in `run'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:253:in `block (2 levels) in execute_action'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/actions/actions_helper.rb:50:in `execute_action'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:231:in `block in execute_action'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:227:in `chdir'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:227:in `execute_action'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:148:in `trigger_action_by_name'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/fast_file.rb:146:in `method_missing'
  Fastfile:23:in `block in parsing_binding'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/lane.rb:33:in `call'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:45:in `chdir'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/runner.rb:45:in `execute'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/lane_manager.rb:54:in `cruise_lane'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/command_line_handler.rb:30:in `handle'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/commands_generator.rb:104:in `block (2 levels) in run'
  /Users/marc.kalmes/gemrepo/gems/commander-fastlane-4.4.5/lib/commander/command.rb:178:in `call'
  /Users/marc.kalmes/gemrepo/gems/commander-fastlane-4.4.5/lib/commander/command.rb:153:in `run'
  /Users/marc.kalmes/gemrepo/gems/commander-fastlane-4.4.5/lib/commander/runner.rb:476:in `run_active_command'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:66:in `run!'
  /Users/marc.kalmes/gemrepo/gems/commander-fastlane-4.4.5/lib/commander/delegates.rb:15:in `run!'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/commands_generator.rb:304:in `run'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/commands_generator.rb:42:in `start'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/fastlane/lib/fastlane/cli_tools_distributor.rb:75:in `take_off'
  /Users/marc.kalmes/gemrepo/gems/fastlane-2.68.2/bin/fastlane:20:in `<top (required)>'
  /Users/marc.kalmes/gemrepo/bin/fastlane:22:in `load'
  /Users/marc.kalmes/gemrepo/bin/fastlane:22:in `<top (required)>'
```
</details>

**Comment by ohayon** [comment]
Hey @mkalmes - thanks for the report. Just for clarification, the newest version of fastlane is in fact returning the proper result? 

If that gym issue is something separate would you mind making a new issue so we can track that problem? Thanks!! 🚀

**Comment by mkalmes** [comment]
Hey @ohayon. That’s correct. The latest release of fastlane works as expected.

The gym issue is documented at #10500 and CocoaPods/Xcodeproj#515.

Thanks for getting back to me! 👍

**Comment by mkalmes** [comment]
Fixed w/ `fastlane` v2.64.0 or newer and `xcodeproj` v1.5.4. 🎉
